onelogin/client.go

```diff
@@ -91,6 +91,11 @@ type GetUserByEmailResponse struct {
 	}
 }
 
+type GenerateSamlAssertionResponseDevices []struct {
```

Comment:
I would re-write this type instead of using it as a sub-component from the `GenerateSamlAssertionResponse` type. Something like this:

```go
// Device represents an MFA device.
type Device struct {
    DeviceID int
    DeviceType string
}
```

Then I would do:

```go
func getDevice(d []*Device) (*Device, error) {...}
```

And then we can also change this
https://github.com/allcloud-io/clisso/blob/0a2f6cf4becb4d1c3b21bf889ddf0729d8532944/onelogin/client.go#L40-L62

...to something like this:

```go
type GenerateSamlAssertionResponse struct {
	Status struct {
		Error   bool   `json:"error"`
		Code    int    `json:"code"`
		Type    string `json:"type"`
		Message string `json:"message"`
	} `json:"status"`
	Data []struct {
		StateToken string `json:"state_token"`
		Devices    []Device {
			DeviceId   int    `json:"device_id"`
			DeviceType string `json:"device_type"`
		}
		CallbackUrl string `json:"callback_url"`
		User        struct {
			Lastname  string `json:"lastname"`
			Username  string `json:"username"`
			Email     string `json:"email"`
			Firstname string `json:"firstname"`
			Id        int    `json:"id"`
		}
	}
}
```